Assets/Scripts/Sharing/AppShareManager.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+using UntitledCube.Maze.Generation;
+using UntitledCube.Timer;
+
+namespace UntitledCube.Sharing
+{
+    public class AppShareManager : MonoBehaviour
```

Comment:
Is there another reason for this to be a monobehaviour except for the start function? if so, instead of setting the listener in the start method you can set it in a method called `static AppShareManager()` which gets called before anything else.

Reply:
Because of the coroutine I'm using, I can use async but that will take me a little while

Reply:
Could you double check if the `WaitForEndOfFrame` is actually needed, because it would be nice if it can just be a static class. If they are needed just leave it as is.

Reply:
If I do that, it doesn't show a picture in the message

Reply:
Then just leave it be.